src/loki/route_action.cc

```diff
@@ -72,14 +72,14 @@ void loki_worker_t::route(Api& request) {
 
   // Validate walking distances (make sure they are in the accepted range)
   if (costing_name == "multimodal" || costing_name == "transit") {
-    auto* ped_opts = options.mutable_costing_options(static_cast<int>(pedestrian));
-    if (!ped_opts->has_transit_start_end_max_distance_case())
-      ped_opts->set_transit_start_end_max_distance(min_transit_walking_dis);
-    auto transit_start_end_max_distance = ped_opts->transit_start_end_max_distance();
+    auto& ped_opts = options.mutable_costing_options()->find(pedestrian)->second;
```

Comment:
there are lots of changes like this because costing options is no longer an array but a map. we can dereference the iterator that comes out of `find` because upstream we make sure the costing is always there in the map